lhotse/recipes/tal_csasr.py

```diff
@@ -0,0 +1,91 @@
+"""
+optional TAL_CSASR(587 hours) if available(https://ai.100tal.com/dataset).
+It is a mandarin-english code-switch corpus.
+"""
+
+import logging
+import os
+import shutil
+import tarfile
+from collections import defaultdict
+from pathlib import Path
+from typing import Dict, Optional, Union
+
+from tqdm.auto import tqdm
+
+from lhotse import validate_recordings_and_supervisions
+from lhotse.audio import Recording, RecordingSet
+from lhotse.supervision import SupervisionSegment, SupervisionSet
+from lhotse.utils import Pathlike, urlretrieve_progress
+
+
+def prepare_tal_csasr(
+    corpus_dir: Pathlike, output_dir: Optional[Pathlike] = None
+) -> Dict[str, Dict[str, Union[RecordingSet, SupervisionSet]]]:
+    """
+    Returns the manifests which consist of the Recordings and Supervisions
+    :param corpus_dir: Pathlike, the path of the data dir.
+    :param output_dir: Pathlike, the path where to write the manifests.
+    :return: a Dict whose key is the dataset part, and the value is Dicts with the keys 'recordings' and 'supervisions'.
+    """
+    corpus_dir = Path(corpus_dir)
+    assert corpus_dir.is_dir(), f"No such directory: {corpus_dir}"
+    if output_dir is not None:
+        output_dir = Path(output_dir)
+        output_dir.mkdir(parents=True, exist_ok=True)
+
+    transcript_path = corpus_dir / "TAL_CSASR" / "label"
+    transcript_dict = {}
+    with open(transcript_path, "r", encoding="utf-8") as f:
+        for line in f.readlines():
+            idx_transcript = line.split()
+            transcript_dict[idx_transcript[0]] = " ".join(idx_transcript[1:])
+
+    manifests = defaultdict(dict)
+    dataset_parts = ["train"]
+    for part in tqdm(
+        dataset_parts, desc="Process tal_csasr audio, it needs waste some time."
+    ):
+        logging.info(f"Processing tal_csasr {part}")
+        # Generate a mapping: utt_id -> (audio_path, audio_info, speaker, text)
+        recordings = []
+        supervisions = []
+        wav_path = corpus_dir / "TAL_CSASR" / "cs_wav"
+        for audio_path in wav_path.rglob("**/*.wav"):
+            logging.info(f"Processing audio path {audio_path}")
+            idx = audio_path.stem
+            speaker = idx
+            if idx not in transcript_dict:
+                logging.warning(f"No transcript: {idx}")
+                continue
+            text = transcript_dict[idx]
+            if not audio_path.is_file():
+                logging.warning(f"No such file: {audio_path}")
+                continue
+            recording = Recording.from_file(audio_path)
```

Comment:
You can speed this up by factor of `num_jobs` using `Recording.from_dir(wav_path, "*.wav", num_jobs=num_jobs)` to parallelize. You will still have to loop over transcript files.

Reply:
OK, I will change.

Reply:
Recording don't from_dir method.

Reply:
Right, I meant RecordingSet.from_dir

Reply:
You can see its usage at
https://github.com/lhotse-speech/lhotse/blob/21b5b3f494893acc2b0ded291236194c803b851a/lhotse/recipes/mgb2.py#L148-L150

https://github.com/lhotse-speech/lhotse/blob/21b5b3f494893acc2b0ded291236194c803b851a/lhotse/recipes/mls.py#L101-L106